internal/uvm/wait.go

```diff
@@ -3,17 +3,34 @@
 package uvm
 
 import (
-	"github.com/Microsoft/hcsshim/internal/logfields"
+	"context"
+	"fmt"
+
 	"github.com/sirupsen/logrus"
+
+	"github.com/Microsoft/hcsshim/internal/logfields"
 )
 
 // Wait waits synchronously for a utility VM to terminate.
-func (uvm *UtilityVM) Wait() error {
-	err := uvm.hcsSystem.Wait()
+func (uvm *UtilityVM) Wait() error { return uvm.WaitCtx(context.Background()) }
+
+// Wait waits synchronously for a utility VM to terminate, or the context to be cancelled.
+func (uvm *UtilityVM) WaitCtx(ctx context.Context) (err error) {
+	err = uvm.hcsSystem.WaitCtx(ctx)
 
-	logrus.WithField(logfields.UVMID, uvm.id).Debug("uvm exited, waiting for output processing to complete")
+	e := logrus.WithField(logfields.UVMID, uvm.id)
+	e.Debug("uvm exited, waiting for output processing to complete")
 	if uvm.outputProcessingDone != nil {
-		<-uvm.outputProcessingDone
+		select {
+		case <-uvm.outputProcessingDone:
+		case <-ctx.Done():
+			if err2 := ctx.Err(); err2 != nil {
+				e.WithError(err2).Warning("failed to wait on uvm output processing")
```

Comment:
Can we make sure we only log this in the case where `err != nil`? It seems unnecessary otherwise as we will return an error with the same message.